### ISSUE TITLE
Incompatible architecture (have 'arm64', need 'x86_64) 

### ISSUE THREAD

**Issue by KalenWessel** [metalbear-co/mirrord#114]
### Bug Description

Mirrord v2.1.0 fails to mirror traffic on Mac M1 due to an incompatible mach-o file.

### Steps to Reproduce

1. curl -fsSL https://raw.githubusercontent.com/metalbear-co/mirrord/main/scripts/install.sh | bash
2. mirrord exec node app.js -n my-app -p my-app-8vqpp   

### Backtrace

```shell
dyld[39680]: terminating because inserted dylib '/var/folders/my/c7grkbzs6v54vhcxhszg15cw0000gn/T/libmirrord_layer.dylib' could not be loaded: tried: '/var/folders/my/c7grkbzs6v54vhcxhszg15cw0000gn/T/libmirrord_layer.dylib' (mach-o file, but is an incompatible architecture (have 'arm64', need 'x86_64')), '/usr/local/lib/libmirrord_layer.dylib' (no such file), '/usr/lib/libmirrord_layer.dylib' (no such file), '/private/var/folders/my/c7grkbzs6v54vhcxhszg15cw0000gn/T/libmirrord_layer.dylib' (mach-o file, but is an incompatible architecture (have 'arm64', need 'x86_64')), '/usr/local/lib/libmirrord_layer.dylib' (no such file), '/usr/lib/libmirrord_layer.dylib' (no such file)
dyld[39680]: tried: '/var/folders/my/c7grkbzs6v54vhcxhszg15cw0000gn/T/libmirrord_layer.dylib' (mach-o file, but is an incompatible architecture (have 'arm64', need 'x86_64')), '/usr/local/lib/libmirrord_layer.dylib' (no such file), '/usr/lib/libmirrord_layer.dylib' (no such file), '/private/var/folders/my/c7grkbzs6v54vhcxhszg15cw0000gn/T/libmirrord_layer.dylib' (mach-o file, but is an incompatible architecture (have 'arm64', need 'x86_64')), '/usr/local/lib/libmirrord_layer.dylib' (no such file), '/usr/lib/libmirrord_layer.dylib' (no such file)
[1]    39680 abort      mirrord exec node app.js -n my-app -p my-app-8vqpp
```

### Your operating system and version

OSX 12.3.1

### Your Node version (`node --version`)

v12.22.12

**Comment by aviramha** [comment]
Hey @KalenWessel 👋 
Thanks for reporting this issue!
We believe this is happening because you're running x86 version of Node while your machine is arm64, which is quite odd (it means that your node runs in Rosetta emulator, meaning worse performance. 
We're thinking on how to still make it work in such case (as it'll probably be a common, and sometimes you really don't have native alternative). In the meantime it'll help us a lot if you ran the following and paste it's output here:
`file $(which node)`

**Comment by KalenWessel** [comment]
@aviramha You're right, it is an x86_64 node.

`~/.nvm/versions/node/v12.22.12/bin/node: Mach-O 64-bit executable x86_64`

**Comment by aviramha** [comment]
Great! Thanks. We're working on a solution, will ping you here once we release it! :)

**Comment by aviramha** [comment]
Hey we just released 2.2.0. Please let us know if it solved the issue for you.
